datasource/opennms/src/main/java/org/opennms/oce/datasource/opennms/processors/AlarmTableProcessor.java

```diff
@@ -65,9 +66,14 @@ public void process(String reductionKey, OpennmsModelProtos.Alarm alarm) {
             final AlarmBean alarmBean = OpennmsMapper.toAlarm(alarm);
             alarmHandlers.forEach(h -> {
                 try {
-                    h.onAlarmCreatedOrUpdated(alarmBean);
+                    // Check if alarm is a situation (meaning the situation originally correlated has come round-trip)
+                    if(alarm.getReductionKey().startsWith(IncidentToEvent.SITUATION_UEI)) {
```

Comment:
This will never evaluate to `true` since the situations are removed from the stream earlier. See https://github.com/OpenNMS/oce/blob/master/datasource/opennms/src/main/java/org/opennms/oce/datasource/opennms/OpennmsDatasource.java#L220.